controller_manager/src/controller_manager.cpp

```diff
@@ -1263,6 +1274,41 @@ void ControllerManager::list_controllers_srv_cb(
         cs.required_state_interfaces = state_interface_config.names;
       }
     }
+    // check for chained interfaces
+    for (const auto & interface : cs.required_command_interfaces)
+    {
+      auto index = interface.find('/');
+      auto prefix = interface.substr(0, index);
```

Comment:
may feel like a silly request but I think a helper function `get_chain_prefix(interface)` would help readability a lot especially since substr doesn't help with resources but produces a new object

> Returns a newly constructed [string](https://cplusplus.com/string) object with its value initialized to a copy of a substring of this object.
https://cplusplus.com/reference/string/string/substr/

or simply return two strings, splitting `interface` so you can eliminate the next `substr` call too

Reply:
I add a helper method called `split_command_interface` to parse the interface string